File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/thisExpression/setterInlinedIntoAnonymousClass.java
package test;

public class Manager {
  private boolean mEnabled;

  public boolean getIsEnabled() {
    return mEnabled;
  }

  private void setIsEnabled(boolean enabled) {
    this.mEnabled = enabled;
  }

  interface Callback {
    void onSuccess(boolean b);

    void onFail();
  }

  void register(Callback c) {}

  void make() {
    register(
        new Callback() {
          @Override
          public void onSuccess(boolean b) {
            setIsEnabled(b);
          }

          @Override
          public void onFail() {
            setIsEnabled(false);
          }
        });
  }
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/thisExpression/setterInlinedIntoAnonymousClass.kt
package test

class Manager {
    var isEnabled: Boolean = false
        private set

    interface Callback {
        fun onSuccess(b: Boolean)

        fun onFail()
    }

    fun register(c: Callback?) {}

    fun make() {
        register(
            object : Callback {
                override fun onSuccess(b: Boolean) {
                    isEnabled = b
                }

                override fun onFail() {
                    isEnabled = false
                }
            })
    }
}
